TaskfMRIAnalysis/scripts/makeSubjectTaskSummary.sh

```diff
@@ -0,0 +1,363 @@
+		dof=`cat ${LevelOneFEATDir}/${Analysis}/dof`;
+		if [[ "$Analysis" = "GrayordinatesStats" || "$Analysis" = "ParcellatedStats" ]] ; then
+			${CARET7DIR}/wb_command -cifti-convert -to-nifti ${LevelOneFEATDir}/${Analysis}/res4d.${Extension} $tmpdir/res4d.nii.gz
+		else
+			cp -suv ${LevelOneFEATDir}/${Analysis}/res4d.nii.gz $tmpdir/res4d.nii.gz
+		fi
+		fslmaths $tmpdir/res4d.nii.gz -Tstd -bin $tmpdir/mask.nii.gz;
+		fslmaths $tmpdir/mask.nii.gz -mul $dof $tmpdir/tdof_t1.nii.gz;
+		
+		if [[ "$Analysis" = "GrayordinatesStats" || "$Analysis" = "ParcellatedStats" ]] ; then
+			for nifti_in in $tmpdir/{mask,tdof_t1}.nii.gz; do
+				cifti_template=$( ls ${LevelOneFEATDir}/${Analysis}/pe*.${Extension} | head -1 )
+				cifti_out=$( echo $nifti_in | sed -e "s|nii.gz|${Extension}|" )
+				${CARET7DIR}/wb_command -cifti-convert -from-nifti ${nifti_in} ${cifti_template} ${cifti_out} -reset-timepoints 1 1 
+			done
+		fi
+		# Make Contrasts.txt file in feat directory
+		ContrastNames=`cat ${LevelOneFEATDir}/design.con | grep "ContrastName" | cut -f 2`
+		NumContrasts=`cat ${LevelOneFEATDir}/design.con | grep "ContrastName" | wc -l`
+		touch ${SummaryDirectory}/Contrasts.txt
+
+		copeCounter=1;
+		while [ "$copeCounter" -le "${NumContrasts}" ] ; do
+			Contrast=`echo $ContrastNames | cut -d " " -f $copeCounter`
+			# Contrasts.txt is used to store the contrast names for this analysis
+			# Avoid writing if contrasts have already been written
+			lenContrastTextFile=$(cat ${SummaryDirectory}/Contrasts.txt | wc -l)
+			if [ "$lenContrastTextFile" -lt "$NumContrasts" ]; then
+				echo ${Contrast} >> ${SummaryDirectory}/Contrasts.txt
+			fi
+	
+			# Check if necessary files exist in analysis directory
+			# if files are missing, write cope number and contrast name to log
+			for File in {cope,varcope} ; do
+				cifti_in=${LevelOneFEATDir}/${Analysis}/${File}${copeCounter}.${Extension}
+				shortName=$( echo $cifti_in | sed -e "s|${ResultsFolder}/||" );
+				if [ -e "$cifti_in" ]; then
+					outdir=${SummaryDirectory}/${Analysis}/cope${copeCounter}.feat
+					if [ ! -d "$outdir" ]; then
+						mkdir -pv $outdir;
+					fi
+					cifti_out=${SummaryDirectory}/${Analysis}/cope${copeCounter}.feat/${File}1.${Extension}
+					# create symlink to summary directory
+					cp -suv $cifti_in $cifti_out
+					echo "${Analysis}: NOTE ${File}${copeCounter} ($Contrast) copied from Level1 $shortName" >> ${SummaryDirectory}/TaskfMRIAnalysis.log
+				else
+					# file is missing, write cope number and contrast name to log
+					echo "${Analysis}: NOTE ${File}${copeCounter} ($Contrast) Level1 is missing file $shortName" >> ${SummaryDirectory}/TaskfMRIAnalysis.log
+				fi
+			done
+			cp -v $tmpdir/mask.${Extension} ${SummaryDirectory}/${Analysis}/cope${copeCounter}.feat/
+			cp -v $tmpdir/tdof_t1.${Extension} ${SummaryDirectory}/${Analysis}/cope${copeCounter}.feat/
+			copeCounter=$(($copeCounter+1))
+		done
+		rm -rv $tmpdir
```

Comment:
`-v` is verbose, why do you want removal of the temporaries to be loud?

Reply:
Changed in next commit